met-web/src/services/contactService/index.tsx

```diff
@@ -0,0 +1,52 @@
+import http from 'apiManager/httpRequestHandler';
+import Endpoints from 'apiManager/endpoints';
+import { replaceUrl } from 'helper';
+import { Contact } from 'models/contact';
+
+export const getContact = async (contactId: number): Promise<Contact> => {
+    const url = replaceUrl(Endpoints.Contacts.GET, 'contact_id', String(contactId));
+    if (!contactId || isNaN(Number(contactId))) {
+        return Promise.reject('Invalid Contact Id ' + contactId);
+    }
+    try {
+        const response = await http.GetRequest<Contact>(url);
+        if (response.data.result) {
+            return Promise.resolve(response.data.result);
+        }
+        return Promise.reject(response.data.message ?? 'Failed to fetch contact');
+    } catch (err) {
+        return Promise.reject(err);
+    }
+};
+
+export const getContacts = async (): Promise<Contact[]> => {
+    try {
+        const response = await http.GetRequest<Contact[]>(Endpoints.Contacts.GET_LIST);
+        if (response.data.result) {
+            return Promise.resolve(response.data.result);
+        }
+        return Promise.reject(response.data.message ?? 'Failed to fetch contact');
+    } catch (err) {
+        return Promise.reject(err);
+    }
+};
+
+interface PostContactRequest {
+    name?: string;
```

Comment:
I guess we can make atleast one column mandatory rite?

Reply:
They are.
        name: yup.string().required(),
        role: yup.string().required(),
        phone_number: yup.string().required(),
        email: yup.string().email().required(),
        address: yup.string().required(),
        bio: yup.string().min(20).required(),

Reply:
but the generated type from yup for some reason defines them as optional. As far as the post call is concerned it will always have the required fields above. But the type I guess has to be defined like that, optional